Add TLS13-KDF support for openssl backend

nicholasberlin opened the pull request:
Relevant issue: https://github.com/microsoft/go/issues/1626

Add TLS13-KDF support
- Update golang-fips to [c494c21](https://github.com/golang-fips/openssl/commit/c494c216979153a2fc07afab57c0858f055ccdc0)
- Use new functions in openssl backend
- Add stub functions for other backends
- Minor documentation fix

nicholasberlin commented on patches/0004-Use-crypto-backends.patch:
```diff
@@ -2452,6 +2457,19 @@ index 00000000000000..573896b9c1e6a8
 +// the underlying functions because the TLS 1.3 KDF does not have a standard of
 +// its own.
 +
++func checkFIPS140Only[Hash hash.Hash](h func() Hash, key []byte) error {
```
this function is copy-pasted from `./src/crypto/hkdf/hkdf.go`

any refactoring suggestions? export the function?

qmuntal replied:
`tls13.ExpandLabel` is an internal API that's not reachable by the user, but only by `crypto/tls`. That package only uses FIPS-compliant parameters when calling `tls13.ExpandLabel`, so there is nothing to check here. You can remove this function.

nicholasberlin replied:
thanks, force pushed an update: https://github.com/microsoft/go/pull/1654/commits/c76587805a1d197c3eb8a667229c868c0f4f9196

nicholasberlin commented:
@microsoft-github-policy-service agree company="Elastic"

qmuntal reviewed (changes requested):
Almost good, thanks!

nicholasberlin committed:
Fix minor document typo

nicholasberlin committed:
Add TLS13-KDF support

- Update golang-fips to c494c21
- Use new functions in openssl backend
- Add stub functions for other backends

nicholasberlin commented:
Ok, managed to "rebase".

qmuntal reviewed (approved):
LGTHM, thanks!

dagood commented:
In case that merge commit looks odd to anyone else--rest assured, it's a merge commit. I think I just got bit by a GitHub UI bug that applied the squash-style message to a merge commit. 😕
